spine-ts/spine-pixi/src/Spine.ts

```
@@ -0,0 +1,407 @@
+					Spine.clipper.clipTriangles(this.verticesCache, numFloats, triangles, triangles.length, uvs, this.lightColor, this.darkColor, useDarkColor);
+
+					finalVertices = Spine.clipper.clippedVertices;
+					finalVerticesLength = finalVertices.length;
+
+					finalIndices = Spine.clipper.clippedTriangles;
+					finalIndicesLength = finalIndices.length;
+				} else {
+					const verts = this.verticesCache;
+					for (let v = 2, u = 0, n = numFloats; v < n; v += vertexSize, u += 2) {
+						let tempV = v;
+						verts[tempV++] = this.lightColor.r;
+						verts[tempV++] = this.lightColor.g;
+						verts[tempV++] = this.lightColor.b;
+						verts[tempV++] = this.lightColor.a;
+
+						verts[tempV++] = uvs[u];
+						verts[tempV++] = uvs[u + 1];
+
+						if (useDarkColor) {
+							verts[tempV++] = this.darkColor.r;
+							verts[tempV++] = this.darkColor.g;
+							verts[tempV++] = this.darkColor.b;
+						}
+					}
+					finalVertices = this.verticesCache;
+					finalVerticesLength = numFloats;
+					finalIndices = triangles;
+					finalIndicesLength = triangles.length;
+				}
+
+				if (finalVerticesLength == 0 || finalIndicesLength == 0) {
+					Spine.clipper.clipEndWithSlot(slot);
+					continue;
+				}
+
+				const mesh = this.getMeshForSlot(slot);
+				mesh.zIndex = i;
+				mesh.updateFromSpineData(texture, slot.data.blendMode, slot.data.name, finalVertices, finalVerticesLength, finalIndices, finalIndicesLength, useDarkColor);
+			}
+
+			Spine.clipper.clipEndWithSlot(slot);
+		}
+		Spine.clipper.clipEnd();
+	}
+
+	public setBonePosition(bone: string | Bone, position: IPointData): void {
+		const boneAux = bone;
+		if (typeof bone === "string") {
+			bone = this.skeleton.findBone(bone)!;
+			this.skeleton.findBone;
+			this.skeleton.findIkConstraint;
+			this.skeleton.findPathConstraint;
+			this.skeleton.findSlot;
+			this.skeleton.findTransformConstraint;
+		}
+
+		if (!bone) {
+			console.error(`Cant set bone position! Bone ${String(boneAux)} not found`);
```

Should probably throw an error.